### PR TITLE
add --mmap in llama-bench

### PR DESCRIPTION
**Pull request by NeoZhangJianyu**
add --no-mmap: in some case, mmap will lead to hang issue on SYCL backend.
show "SYCL" backend in result table.
update get_gpu_info() for SYCL.

### CODE REVIEW

File: examples/llama-bench/llama-bench.cpp [hunk 1]
```diff
@@ -120,6 +121,20 @@ static std::string get_gpu_info() {
             id += "/";
         }
     }
+#endif
+#ifdef GGML_USE_SYCL
+    int device_list[GGML_SYCL_MAX_DEVICES];
+    ggml_sycl_get_gpu_list(device_list, GGML_SYCL_MAX_DEVICES);
+
+    for (int i = 0; i < GGML_SYCL_MAX_DEVICES; i++) {
+        if (device_list[i]>0){
```
Comment (slaren) [inline examples/llama-bench/llama-bench.cpp:130]:
```suggestion
        if (device_list[i] > 0){
```

File: examples/llama-bench/llama-bench.cpp [hunk 2]
```diff
@@ -201,6 +218,7 @@ static void print_usage(int /* argc */, char ** argv) {
     printf("  -sm, --split-mode <none|layer|row>  (default: %s)\n", join(transform_to_str(cmd_params_defaults.split_mode, split_mode_str), ",").c_str());
     printf("  -mg, --main-gpu <i>                 (default: %s)\n", join(cmd_params_defaults.main_gpu, ",").c_str());
     printf("  -nkvo, --no-kv-offload <0|1>        (default: %s)\n", join(cmd_params_defaults.no_kv_offload, ",").c_str());
+    printf("  -nmmap, --no-mmap                   (default: %s)\n", cmd_params_defaults.use_mmap ? "0" : "1");
```
Comment (slaren) [inline examples/llama-bench/llama-bench.cpp:221]: Print the possible values (`<0|1>`). The parameter is `no-mmap` but the value is `mmap`, the discrepancy needs to be fixed.

File: examples/llama-bench/llama-bench.cpp [hunk 3]
```diff
@@ -161,6 +176,7 @@ struct cmd_params {
     std::vector<bool> no_kv_offload;
     std::vector<bool> mul_mat_q;
     std::vector<std::vector<float>> tensor_split;
+    bool use_mmap;
```
[the remark this answers is not in the document]
Comment (NeoZhangJianyu) [reply examples/llama-bench/llama-bench.cpp:179]: use_mmap is a fixed parameter to resolve SYCL memcpy hang issue in some cases.
No need to test with different value.

[the remark this answers is not in the document]
Comment (slaren) [reply examples/llama-bench/llama-bench.cpp:179]: I understand that is the purpose of this change, but it is still useful to compare the performance with and without mmap in some cases. Every other model parameter can be specified multiple times, and there is no reason to treat this differently.

File: examples/llama-bench/llama-bench.cpp [hunk 4]
```diff
@@ -120,6 +121,20 @@ static std::string get_gpu_info() {
...
+
+    for (int i = 0; i < GGML_SYCL_MAX_DEVICES; i++) {
+        if (device_list[i]>0){
+            char buf[128];
+            ggml_sycl_get_device_description(i, buf, sizeof(buf));
+            id += buf;
+            id += "/";
+        }
+    }
+    if(id.length()>2) id.pop_back();
```
[the remark this answers is not in the document]
Comment (NeoZhangJianyu) [reply examples/llama-bench/llama-bench.cpp]: Yes, accept.

File: examples/llama-bench/llama-bench.cpp (hunk 1 above)
Comment (NeoZhangJianyu) [reply examples/llama-bench/llama-bench.cpp]: Yes, accept.

File: examples/llama-bench/llama-bench.cpp [hunk 5]
```diff
@@ -490,6 +512,7 @@ struct cmd_params_instance {
         cparams.mul_mat_q = mul_mat_q;
         cparams.offload_kqv = !no_kv_offload;
 
+
```
[the remark this answers is not in the document]
Comment (NeoZhangJianyu) [reply examples/llama-bench/llama-bench.cpp]: Yes, accept.

**Comment by NeoZhangJianyu** [comment]
@airMeng , @luoyu-intel , @abhilash1910, @ggerganov Invite you to review if you are idle.

**Review by slaren — CHANGES_REQUESTED** [review]
- Change the mmap parameter name from `no-mmap` to `mmap`. Set it by default to 1.
- Allow multiple values
- Print the value in the markdown printer if it is not the default.

**Comment by NeoZhangJianyu** [comment]
> * Change the mmap parameter name from `no-mmap` to `mmap`. Set it by default to 1.
> * Allow multiple values
> * Print the value in the markdown printer if it is not the default.

OK, I will update according to comments.

**Review by slaren — APPROVED** [review]

File: examples/llama-bench/llama-bench.cpp [hunk 6]
```diff
@@ -508,6 +539,7 @@ static std::vector<cmd_params_instance> get_cmd_params_instances(const cmd_param
     for (const auto & tv : params.type_v)
     for (const auto & mmq : params.mul_mat_q)
     for (const auto & nkvo : params.no_kv_offload)
+    for (const auto & mmp : params.use_mmap)
```
Comment (slaren) [inline examples/llama-bench/llama-bench.cpp:542]: One minor thing, move this up, next to the other model parameters (after `tensor_split` for example), to minimize the number of times the model needs to be reloaded.

Comment (NeoZhangJianyu) [reply examples/llama-bench/llama-bench.cpp:542]: Yes.

**Comment by slaren** [comment]
The changes to `llama-bench` are good, should we merge this now, or wait for a review of the ggml-sycl changes?

**Review by abhilash1910 — APPROVED** [review]

**Pull request merged**